*** a/scripts/gen_actor_source.py
--- b/scripts/gen_actor_source.py
***************
*** 1013,1019 ****
          :return:            Kumir module object
          """
          f = open(file_name, 'r', encoding="utf-8")
!         data = json.load(f, encoding="utf-8")
          f.close()
          absolute_path = os.path.abspath(file_name)
          module_dir = os.path.dirname(absolute_path)
--- 1013,1019 ----
          :return:            Kumir module object
          """
          f = open(file_name, 'r', encoding="utf-8")
!         data = json.load(f)
          f.close()
          absolute_path = os.path.abspath(file_name)
          module_dir = os.path.dirname(absolute_path)
